lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -727,7 +732,11 @@ def rank_ecc(
         for rawfc, calfc in zip(
                 raw_forecast_members.slices_over("time"),
                 post_processed_forecast_percentiles.slices_over("time")):
-            random_data = np.random.random(rawfc.data.shape)
+            if fixed_random_seed is not None:
```

Comment:
That's clever. I didn't realise the behaviour of `np.random.RandomState` is it was passed `None`.